### ISSUE TITLE
SteamOS iso USB install wipes all partitions on first drive without prompt

### ISSUE THREAD

**Issue by 256colour** [ValveSoftware/SteamOS#668]
#### Your system information

* Steam client version: n/a
* SteamOS version: DVD iso installer dated 2018-01-22 14:42 | 1.9G (located here)[
http://repo.steampowered.com/download/SteamOSDVD.iso]

#### Please describe your issue in as much detail as possible:
What 'should' happen:
- A USB installer *should* prompt the user to either automatically install the OS, or install prompt-by-prompt in 'expert' mode.
- A USB installer *should* prompt the user before making DESTRUCTIVE ALTERATIONS to their hard drive.

What happened instead:
- Wrote SteamOSDVD.iso to a usb stick, and booted it on my 2011 iMac.
- Installer sat for 45 seconds at black screen, no prompt.
- Suddenly installer begins loading SteamOS installer. Great! Great? Great.........
- SteamOS installer begins scanning my hardware and loading itself into memory? I definitely did not press any buttons to indicate I was ready to install....
- SteamOS installer begins scanning my drives to determine which disks it should... oh my god...
- SteamOS installer begins re-partitioning my drives, and trying to rewrite the system EFI.
- SteamOS installer fails to write the EFI. This is where I finally get control of my machine again, and survey the damage.
- One year of development code on my OS X, Windows 7, and Ubuntu drives are gone thanks to the installer rewriting my primary disk's partition table without asking permission to.

#### Steps for reproducing this issue:

Hardware:
- 2011 Apple iMac 3.1 GHz i5, 12GB ram, ATI Radeon 6970m, 1.25 TB SSD+HD fusion drive
- 32gb USB stick

Software:
- (DVD iso installer dated 2018-01-22 14:42)[
http://repo.steampowered.com/download/SteamOSDVD.iso]
- Startup Disk Creator on Ubuntu 18.04 Desktop to write the ISO to a USB stick

1. Write ISO to usb stick using Startup Disk Creator
2. Boot iMac while holding left-Shift key to get boot prompt. Select USB stick.
3. Wait for 30 seconds - 1 minute at black screen. Not sure if this is grub trying to find a system to boot?
4. SteamOS installer automatically begins installing and wipes `/dev/sda`, writing its own partitions wherever it wishes.

Obviously I am unbelievably and correctly angry about an installer capable of such destructive data writes without a single prompt. I do not even know what to suggest here, as I've never seen an installer in 35 years that did not prompt at least once to make sure the user really wanted to wipe a drive.

**Comment by TTimo** [comment]
Hello @256colour, we're sorry this happened to you.

This is the first screen that comes up when booting from the installer: https://www.dropbox.com/s/thod69cib8360ss/SteamOS%202%20.148%20installer.jpg?dl=0

Is it possible this screen didn't display on your hardware? The prompt is not on a timeout, it will not go into the Automated Install without a key press, but if the display failed to initialize I can see how that would be a problem.

**Comment by krisman** [comment]
@256colour Are you using rEFInd, by any chance? We observed a bug in it, that made it select the first boot option, Automatic Install.

**Comment by 256colour** [comment]
@krisman I don't believe so. I believe I was running from vanilla OS X with Boot camp installed.
